insert.js

```diff
@@ -10,14 +10,20 @@ platform.core.node({
   controlOutputs: ['no_connection', 'bad_input'],
 }, (inputs, output, control) => {
   if (instance) {
+    let data = inputs.data;
+    data = JSON.parse(
+      JSON.stringify(Object.assign({}, inputs.data))
+    );
```

Comment:
this is costly on performance, what is the aim here?

Reply:
It doesn't accept nested objects for some reason.

> Error: Argument "data" is not a valid Document. Invalid use of type "object" as a Firestore argument.
    at Validator.<computed> [as isDocument]

Reply:
yeah its not a good hack. you are affecting EVERY possible insert. apply the hack to your intended objects before passing them to this package.

Reply:
I tried that, for some reason even though the objects look identical when outputting them, it results in an error when I use the one passed through the parameters, but it works when the stringify/parse are done directly after the call

Reply:
p.s. if you are sure this is an issue with the firestore package, please open an issue on the [corresponding repository](https://github.com/googleapis/nodejs-firestore) and using the temporary hack in the consumer code instead of bringing it to this package. however, I would highly recommend updating the package `@google-cloud/firestore` and testing with the updated version beforehand.

Reply:
It is with the way the platform is handling objects somehow. Putting the plain javascript object directly in the file works.

Reply:
> I tried that, for some reason even though the objects look identical when outputting them, it results in an error when I use the one passed through the parameters, but it works when the stringify/parse are done directly after the call

then the objects clearly are not identical. anyways, regardless, this is either an issue with the consumer code or firestore package, and in either case the fix/hack should not be applied to this package.

Reply:
> It is with the way CONNECT platform is handling objects somehow. Putting the plain javascript object directly in the file works.

in that case the issue should be fixed on CONNECT-platform.

Reply:
I know that they are not identical. But from a text dump perspective, they are. So the differences have to be checked through other ways. Either way, my guess is that this lies in the way CONNECT platform is handling object parameters (as already mentioned).

Reply:
maybe looking at these help with finding the source of the issue:

https://github.com/googleapis/nodejs-firestore/blob/d36f5092fcdee0fb779b6fa51d7da6534ca76e36/dev/src/validate.ts#L126-L136

https://github.com/googleapis/nodejs-firestore/blob/d36f5092fcdee0fb779b6fa51d7da6534ca76e36/dev/src/util.ts#L78-L80

Reply:
It would be helpful to know how CONNECT platform treats inputs especially when there are nested objects

Reply:
@loreanvictor even without considering the conversion it still is the case that the two objects are somehow different

Object after going through stringify and parse in the call function
`{
  content: 'test post',
  created_at: '2020-02-14T12:10:32.848Z',
  user: {
    _id: 'LzBxBKx6CpvvUBlx4XIJ',
    owner_id: '9oX8FQ3Hac6C2IH7IK14',
    email: 'mustapha3892@gmail.com',
    name: 'linkedin',
    collection: 'companies',
    created_at: '2020-02-09T16:17:51.631Z'
  },
  link: {
    has: true,
    url: 'https://connect-platform.com/',
    title: 'Connect Platform - Visual Backend Programming For Programmers',
    description: 'Blazing-fast cloud-native visual programming, same low-level control of traditional coding, and seamless cloud deployments. Embrace the future of web development.',
    mediaType: 'website',
    image: 'https://connect-platform.com/svg/logo.svg'
  }
}`

Object after going through stringify and parse in the node code and then passed as an input
`{
  content: 'test post',
  created_at: '2020-02-14T12:10:32.848Z',
  user: {
    _id: 'LzBxBKx6CpvvUBlx4XIJ',
    owner_id: '9oX8FQ3Hac6C2IH7IK14',
    email: 'mustapha3892@gmail.com',
    name: 'linkedin',
    collection: 'companies',
    created_at: '2020-02-09T16:17:51.631Z'
  },
  link: {
    has: true,
    url: 'https://connect-platform.com/',
    title: 'Connect Platform - Visual Backend Programming For Programmers',
    description: 'Blazing-fast cloud-native visual programming, same low-level control of traditional coding, and seamless cloud deployments. Embrace the future of web development.',
    mediaType: 'website',
    image: 'https://connect-platform.com/svg/logo.svg'
  }
}`

> typeof data object
typeof data.user object
toString data in call[object Object]
toString data.user in call[object Object]
toString data.user._id in call[object String]

> typeof inputs.data object
typeof inputs.data.user object
toString inputs.data in call[object Object]
toString inputs.data.user in call[object Object]
toString inputs.data.user._id in call[object String]

Reply:
Looks like the package is way out of date and updating to the latest firestore nodejs package did the trick